File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.size() > t.size()) return false;
        if(!s.size()) return true;
        
        int p1 = 0;
        int p2 = 0;
        
        while(p2 < t.size()) {
            
            if(s[p1] == t[p2]) {
                p1++;
                p2++;
                if(p1 == s.size()) return true;
            } else {
                p2++;
            }
            
        }
        
        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.ts
function isSubsequence(s: string, t: string): boolean {
    if (s.length > t.length) return false;
    if (s.length === 0) return true;

    let p1 = 0;
    let p2 = 0;

    while (p2 < t.length) {
      if (s[p1] === t[p2]) {
        p1++;
        if (p1 === s.length) return true;
      }
      p2++;
    }

    return false;
};